### ISSUE TITLE
Get stuck when running dataset=DVS128Gesture(dataset_dir, train=False, data_type='frame', frames_number=30000, split_by == 'time')

### ISSUE THREAD

**Issue by Ikarosy** [fangwei123456/spikingjelly#123]
I tried the code below for 5 times, it always gets stuck when intergrating the same [events_np/train/5/user03_fluorescent_0.npz] into [frames_number_30000_split_by_time/train/5] and stops proceeding.
> dataset=DVS128Gesture(dataset_dir, train=False, data_type='frame', duration=30000, split_by = 'time')

Then, I tried without split_by == 'time'
> dataset=DVS128Gesture(dataset_dir, train=False, data_type='frame', duration=30000) 

it finished the whole converting process but failed as the following assertion goes:

> /anaconda3/lib/python3.8/site-packages/spikingjelly/datasets/__init__.py in __init__(self, root, train, data_type, frames_number, split_by, duration, transform, target_transform)
>     671             if frames_number is not None:
>     672                 assert frames_number > 0 and isinstance(frames_number, int)
> --> 673                 assert split_by == 'time' or split_by == 'number'
>     674                 frames_np_root = os.path.join(root, f'frames_number_{frames_number}_split_by_{split_by}')
>     675                 if os.path.exists(frames_np_root):

**Comment by Ikarosy** [comment]
create the file duration_30000 via 
> dataset=DVS128Gesture(dataset_dir, train=False, data_type='frame', duration=30000)

then rerun via 

> dataset=DVS128Gesture(dataset_dir, train=False, data_type='frame', duration=30000, split_by = 'time')

will solve the issue

**Comment by Ikarosy** [comment]
and there are some bugs remained unsolved in 
`from spikingjelly.datasets import pad_sequence_collate
from spikingjelly.datasets.dvs128_gesture import DVS128Gesture

train_set = DVS128Gesture(args.data_dir, train=True, data_type='frame', duration=args.duration)
    train_data_loader = DataLoader(
        dataset=train_set,
        collate_fn=pad_sequence_collate,
        batch_size=args.b,
        shuffle=True,
        num_workers=args.j,
        drop_last=True,
        pin_memory=True)
`
the tensors' sizes are not identical, so the stack won't work

**Comment by fangwei123456** [comment]
Thanks for your bug report. I will check it.

**Comment by fangwei123456** [comment]
There is no error in the following codes:
```python
from spikingjelly.datasets import dvs128_gesture

dts = dvs128_gesture.DVS128Gesture(root='D:/datasets/DVS128Gesture', train=False, data_type='frame', duration=30000)
print(dts[0])

```

It is a wrong way to create frames in the following codes:
```python
dataset=DVS128Gesture(dataset_dir, train=False, data_type='frame', duration=30000, split_by = 'time')
```

![image](https://user-images.githubusercontent.com/38352109/134833629-d6bebf5d-fc8b-47ef-a03e-7086903ec3d8.png)

`split_by` is used for fixed frames number.  `duration` is used for fixed time duration for each frame.

**Comment by fangwei123456** [comment]
```python
from spikingjelly.datasets import pad_sequence_collate
from spikingjelly.datasets.dvs128_gesture import DVS128Gesture
from torch.utils.data import DataLoader

if __name__=='__main__':
    train_set = DVS128Gesture(root='D:/datasets/DVS128Gesture', train=True, data_type='frame', duration=1000000)
    train_data_loader = DataLoader(
        dataset=train_set,
        collate_fn=pad_sequence_collate,
        batch_size=16,
        shuffle=True,
        num_workers=4,
        drop_last=True,
        pin_memory=True)

    for x, y, x_len in train_data_loader:
        print(x.shape)
        print(y.shape)
        print(x_len)
        exit()
```

Outputs are:

```
The directory [D:/datasets/DVS128Gesture\duration_1000000] already exists.
torch.Size([16, 9, 2, 128, 128])
torch.Size([16])
tensor([7, 7, 8, 6, 7, 6, 7, 7, 6, 9, 5, 8, 8, 6, 6, 7])
```

**Comment by fangwei123456** [comment]
The above codes are running in the latest version:
https://github.com/fangwei123456/spikingjelly/tree/7c4d739c3f99b25182f72723626db1732fb58deb

**Comment by Ikarosy** [comment]
Thanks for your reply.  I see it is a wrong way to run `dataset=DVS128Gesture(dataset_dir, train=False, data_type='frame', duration=30000, split_by = 'time')`, then I tried `dataset=DVS128Gesture(dataset_dir, train=False, data_type='frame', duration=30000)`. and it works well.

**Comment by fangwei123456** [comment]
You are welcome! I notice that your SJ codes have some different with the latest version.
Yours:

`674 frames_np_root = os.path.join(root, f'frames_number_{frames_number}split_by{split_by}')`


The latest in Github:
`frames_np_root = os.path.join(root, f'frames_number_{frames_number}_split_by_{split_by}')`

You can try to update SJ by 'git pull', then delete all files except for `download` in `dataset_dir`.

The `download` dir is:
```
.
|-- DvsGesture.tar.gz
|-- LICENSE.txt
|-- README.txt
`-- gesture_mapping.csv
```


Then you can try to run codes again, and SJ will create datasets from the begining.
